doc/contribute/guidelines.rst

```diff
@@ -536,12 +536,28 @@ reference manuals, etc.
 Coding Style
 ============
 
+.. note::
+   Coding style is enforced on any new or modified code, but contributors are
+   not expected to correct the style on existing code that they are not
+   modifying.
+
+.. note::
+   For style aspects where the guidelines don't offer explicit guidance or
+   permit multiple valid ways to express something, new code must try to follow
+   the style of existing code in the tree, with higher importance given to
+   "nearby" code (first look at the function, then the same file, then
+   subsystem, etc).
+
 .. _Linux kernel coding style:
    https://kernel.org/doc/html/latest/process/coding-style.html
 
+.. _snake case:
+   https://en.wikipedia.org/wiki/Snake_case
+
 In general, follow the `Linux kernel coding style`_, with the following
-exceptions:
+exceptions and clarifications:
 
+* Use `snake case`_ for code and variables.
 * The line length is 100 columns or fewer. In the documentation, longer lines
   for URL references are an allowed exception.
```

Comment:
Given there are quite many places where line length is over 100 columns, does the rule "try to align with style in nearby code" apply? I.e. we're now saying we're not expecting contributors to fix non-conforming code/files, so do we want them to introduce potentially oddly/inconsistently wrapped content to existing files?

Reply:
Up to others to comment on this, but for me this question really falls on the "case by case" and "common sense" criteria of the reviewers. I don't think rewriting the text or complicating it further will help solve all the corner cases like this one.

Reply:
New code should comply with the requirements

Reply:
In principle I do not think this text needs changing, but ping me if you disagree @kartben.